NYPLAudiobookToolkit/UI/AudiobookPlayerViewController.swift

```diff
@@ -596,30 +710,47 @@ let SkipTimeInterval: Double = 15
     }
 }
 
-extension AudiobookPlayerViewController: AudiobookTableOfContentsTableViewControllerDelegate {
-    public func userSelectedSpineItem(item: SpineElement) {
-
-        self.waitingForPlayer = true
-        self.activityIndicator.startAnimating()
-
-        self.playbackControlView.showPauseButtonIfNeeded()
-
-        let selectedChapter = item.chapter
-        let timeLeftInBook = self.timeLeftAfter(chapter: selectedChapter)
-        self.seekBar.setOffset(
-            selectedChapter.playheadOffset,
-            duration: selectedChapter.duration,
-            timeLeftInBook: timeLeftInBook,
-            middleText: self.middleTextFor(chapter: selectedChapter)
-        )
-
-        if self.audiobookManager.audiobook.player.isPlaying {
-            self.shouldBeginToAutoPlay = true
-        } else {
-            self.shouldBeginToAutoPlay = false
-        }
-        self.navigationController?.popViewController(animated: true)
-    }
+extension AudiobookPlayerViewController: AudiobookReaderPositionSelectionDelegate {
+  func didSelectTOC(_ spineElement: SpineElement) {
+    advancePlayer(to: spineElement.chapter)
+    self.navigationController?.popViewController(animated: true)
+  }
+  
+  func didSelectBookmark(_ bookmark: NYPLAudiobookBookmark) {
+    guard let chapterLocation = ChapterLocation(number: bookmark.chapter,
+                                                part: bookmark.part,
+                                                duration: bookmark.duration,
+                                                startOffset: 0,
+                                                playheadOffset: bookmark.time,
+                                                title: bookmark.title,
+                                                audiobookID: bookmark.audiobookId) else {
+      return
+    }
+    advancePlayer(to: chapterLocation)
+    self.navigationController?.popViewController(animated: true)
+  }
+  
+  func advancePlayer(to chapter: ChapterLocation) {
```

Comment:
@ettore This function is also from `userSelectedSpineItem(item:)`, I refactored into a separated function so I can reuse it for bookmark and chapter selection.